app/editor/src/components/navbar/NavBarGroup.tsx

```diff
@@ -0,0 +1,16 @@
+import * as styled from './NavBarStyled';
+
+export interface INavBarGroupProps {
+  /**
+   * choose a background colour for the navigation bar
+   */
+  backgroundColor?: string;
+  /**
+   * place desired nodes within the navigation bar group
+   */
+  children?: React.ReactNode;
+}
+
+export const NavBarGroup: React.FC<INavBarGroupProps> = ({ children }) => {
```

Comment:
Just inherit from the appropriate element to get all normal properties.  Don't do a background color prop, instead use a styled property theme.